relay-wstring/src/lib.rs

```diff
@@ -380,10 +408,10 @@ impl fmt::Display for WStr {
 /// These surrogate code units have the first 6 bits set to a fixed prefix identifying
 /// whether they are the *leading* or *trailing* code unit of the surrogate pair.  And for
 /// the leading surrogate this bit prefix is `110110`, thus all leading surrogates have a
-/// code unit between 0xD800-0xDBFF.
+/// code unit of `0xD8xx`.
 #[inline]
 fn is_leading_surrogate(code_unit: u16) -> bool {
-    code_unit & 0xD800 == 0xD800
+    code_unit & 0xFC00 == 0xD800
```

Comment:
Turns out the issue was in `is_leading_surrogate` and `is_trailing_surrogate`. Instead of checking the entire byte to match for `0xD8`, we only checked some bits. This would lead match too many patterns, for instance: `0xF8` (`0b0111`) has all bits set of `0xD8` (`0b0110`) but is outside of the surrogate range.